packages/svelte/src/compiler/types/template.d.ts

```diff
@@ -457,6 +457,11 @@ export type Block = EachBlock | IfBlock | AwaitBlock | KeyBlock | SnippetBlock;
 export interface Attribute extends BaseNode {
 	type: 'Attribute';
 	name: string;
+	/**
+	 * The attribute may be omitted when false.
+	 * String values are represented by an array since it may be a combination of text and expression
+	 * values such as `style="color: {color} !import"`.
```

Comment:
not sure why we would need to specify 'attribute may be omitted when false' — for one thing non-boolean attributes are never `false`, they either exist or they don't

```suggestion
	 * Quoted/string values are represented by an array, even if they contain a single expression like `"{x}"`
```

Reply:
> not sure why we would need to specify 'attribute may be omitted when false' — for one thing non-boolean attributes are never false, they either exist or they don't

because not everyone knows that. I had no clue that was the case :smile:

> Quoted/string values are represented by an array, even if they contain a single expression like `"{x}"`

The part that I was trying to describe was what the array represented. I spent awhile looking at it thinking it was holding multiple values (e.g. maybe the same attribute was declared twice with different values)